client/rpc/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+    T: Send + Sync + TrieLayout + 'static,
+{
+    async fn load_file_in_storage(
+        &self,
+        file_path: String,
+        location: String,
+        owner: AccountId32,
+    ) -> RpcResult<FileMetadata> {
+        // Open file in the local file system.
+        let mut file = File::open(PathBuf::from(file_path.clone())).map_err(into_rpc_error)?;
+
+        // Instantiate an "empty" [`FileDataTrie`] so we can write the file chunks into it.
+        let mut file_data_trie = <FL as FileStorage<T>>::FileDataTrie::default();
+        // A chunk id is simply an integer index.
+        let mut chunk_id: u64 = 0;
+
+        // Read file in chunks of [`FILE_CHUNK_SIZE`] into buffer then push buffer into a vector.
+        // Loops until EOF or until some error that is NOT `ErrorKind::Interrupted` is found.
+        // If `ErrorKind::Interrupted` is found, the operation is simply retried, as per
+        // https://doc.rust-lang.org/std/io/trait.Read.html#errors-1
+        loop {
+            let mut chunk = Vec::with_capacity(FILE_CHUNK_SIZE);
+            let read_result = file
+                .by_ref()
+                .take(FILE_CHUNK_SIZE as u64)
+                .read_to_end(&mut chunk);
+            match read_result {
+                // Reached EOF, break loop.
+                Ok(0) => {
+                    debug!(target: LOG_TARGET, "Finished reading file");
+                    break;
+                }
+                // Haven't reached EOF yet, continue loop.
+                Ok(bytes_read) => {
+                    debug!(target: LOG_TARGET, "Read {} bytes from file", bytes_read);
+
+                    // Build the actual [`FileDataTrie`] by inserting each chunk into it.
+                    file_data_trie
+                        .write_chunk(&chunk_id, &chunk)
+                        .map_err(into_rpc_error)?;
+                    chunk_id += 1;
+                }
+                Err(e) => {
+                    error!(target: LOG_TARGET, "Error when trying to read file: {:?}", e);
+                    return Err(into_rpc_error(e));
+                }
+            }
+        }
+
+        // Generate the necessary metadata so we can insert file into the File Storage.
+        let root = file_data_trie.get_root();
+        let fs_metadata = file.metadata().map_err(into_rpc_error)?;
+
+        // Build StorageHub's [`FileMetadata`]
+        let file_metadata = FileMetadata {
+            size: fs_metadata.len(),
+            fingerprint: root.as_ref().into(),
+            owner: owner.to_string(),
+            location: location.clone().into(),
+        };
+        let file_key = file_metadata.key::<T::Hash>();
+
+        // Acquire FileStorage write lock.
+        let mut file_storage_write_lock = self.file_storage.write().await;
+
+        // Finally store file in File Storage.
+        file_storage_write_lock
+            .insert_file_with_data(file_key, file_metadata.clone(), file_data_trie)
```

Comment:
So I only realized what this method was actually trying to do until I reached the end. I think we should rename this method to `upload_file_to_storage`.